libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Notifications.h

```diff
@@ -0,0 +1,119 @@
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#import <Foundation/Foundation.h>
+#import "SFRestAPI.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SFRestAPI(Notifications)
+
+/**
+ * Returns a request to fetch the status of notifications, including unread and unseen count.
+ * @param apiVersion API version.
+ */
+- (SFRestRequest *)requestForNotificationsStatus:(NSString *)apiVersion;
+
+/**
+ * Returns a request to fetch the given notification.
+ * @param notificationId ID of notification to fetch.
+ * @param apiVersion API version.
+ */
+- (SFRestRequest *)requestForNotification:(NSString *)notificationId apiVersion:(NSString *)apiVersion;
+@end
+
+NS_SWIFT_NAME(FetchNotificationsRequestBuilder)
+@interface SFSDKFetchNotificationsRequestBuilder: NSObject
+
+/**
+ * Sets the number of notifications to fetch. Max and default are 20.
+ * @param size Number of notifications to fetch.
+ */
+- (SFSDKFetchNotificationsRequestBuilder *)setSize:(NSUInteger)size;
+
+/**
+ * Notifications occurring before the provided date will be fetched. Shouldn't be used with `setAfter`.
+ * @param date ISO 8601 formatted string.
+ */
+- (SFSDKFetchNotificationsRequestBuilder *)setBefore:(NSString *)date;
+
+/**
+ * Notifications occurring after the provided date will be fetched. Shouldn't be used with `setBefore`.
+ * @param date ISO 8601 formatted string.
+ */
+- (SFSDKFetchNotificationsRequestBuilder *)setAfter:(NSString *)date;
+
+/**
+ * Returns a request to fetch notifications based on values from the builder.
+ * @param apiVersion API version.
+ */
+- (SFRestRequest *)buildFetchNotificationsRequest:(NSString *)apiVersion;
+
+@end
+
+NS_SWIFT_NAME(UpdateNotificationsRequestBuilder)
+@interface SFSDKUpdateNotificationsRequestBuilder: NSObject
+
+/**
+ * Sets the notification to update. Shouldn't be used with `setNotificationIds` or `setBefore`.
+ * @param notificationId ID of notification to update.
+ */
+- (SFSDKUpdateNotificationsRequestBuilder *)setNotificationId:(NSString *)notificationId;
+
+/**
+ * Sets a list of notifications to update (max 50). Shouldn't be used with `setNotificationId` or `setBefore`.
+ * @param notificationIds Array of notifications IDs to update.
+ */
+- (SFSDKUpdateNotificationsRequestBuilder *)setNotificationIds:(NSArray<NSString *> *)notificationIds;
+
+/**
+ * Notifications occurring before the provided date will be updated. Shouldn't be used with `setNotificationId` or `setNotificationIds`.
+ * @param date ISO 8601 formatted string.
+ */
+- (SFSDKUpdateNotificationsRequestBuilder *)setBefore:(NSString *)date;
```

Comment:
The method should take a NSDate object and the builder should be the one in charge of generating the correct formatted string.